### PR TITLE
Cu 8693z4817 fix signature

### PR DESCRIPTION
**Pull request by Woody1193**
This PR fixes an issue wherein requests sent to the MMS were being rejected because they were not encoded properly. Additionally, this PR refactors requests to be multi-part.

### CODE REVIEW

**Comment by Woody1193** [comment]
Task linked: [CU-8693z4817 MMS API service](https://app.clickup.com/t/8693z4817)

**Review by ricardomartinezmontero — APPROVED** [review]

File: src/mms_client/utils/multipart_transport.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+
+def get_boundary() -> str:
+    """Return a randomized MIME boundary string."""
+    return f"MIMEBoundary_{now_b64()}".center(33, "=")
+
+
+def get_content_id() -> str:
+    """Return a randomized MIME content ID string."""
+    return f"<{now_b64()}@electroroute.co.jp>"
```
Comment (Liam-Woods) [inline src/mms_client/utils/multipart_transport.py:57]: Get rid of this!

File: src/mms_client/utils/multipart_transport.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+
+def overwrie_attachnode(node):
```
Comment (Liam-Woods) [inline src/mms_client/utils/multipart_transport.py:60]: typo

File: src/mms_client/utils/multipart_transport.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    Returns:    The MIME multipart object.
+    """
+    part = MIMEMultipart(
+        "related", charset="UTF-8", type="application/xop+xml", boundary=get_boundary(), start=content_id
+    )
+    part.set_param("start-info", "application/soap+xml")
+    return part
+
+
+def get_envelopepart(envelope: Element, content_id: str) -> MIMEApplication:
```
Comment (Liam-Woods) [inline src/mms_client/utils/multipart_transport.py:89]: get_envelope_part

File: src/mms_client/utils/multipart_transport.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    def __init__(self, name: str, data: bytes):
+        """Create a new attachment.
+
+        Arguments:
+        name (str):   The name of the attachment.
+        data (bytes): The data to be attached.
+        """
+        self.name = name
+        self.data = data
+        self.cid = f"{get_id()}-{name}@electroroute.co.jp"
```
Comment (Liam-Woods) [inline src/mms_client/utils/multipart_transport.py:118]: Another one

File: src/mms_client/utils/multipart_transport.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        self.tag = FILETAG + self.cid
+
+
+class MultipartTransport(Transport):
+    """A transport that supports MTOMS attachments."""
```
Comment (Liam-Woods) [inline src/mms_client/utils/multipart_transport.py:123]: What is MTOMS? Is this a term that someone familiar with MMS would also know? Would this benefit from an explainer in the README?

**Review by Liam-Woods — APPROVED** [review]

File: src/mms_client/utils/multipart_transport.py (hunk 1 above)
Comment (Woody1193) [reply src/mms_client/utils/multipart_transport.py]: It has been gotten rid of.

File: src/mms_client/utils/multipart_transport.py (hunk 3 above)
Comment (Woody1193) [reply src/mms_client/utils/multipart_transport.py]: I can see the underscore is a bit weak with this one

File: src/mms_client/utils/multipart_transport.py (hunk 5 above)
Comment (Woody1193) [reply src/mms_client/utils/multipart_transport.py:130]: MTOM stands for "Message Transmission Optimization Mechanism" and it is used when sending XOP (XML-binary optimized protocol) messages. Basically, this is restricted to SOAP multi-part requests. Please forget everything you read now.

File: src/mms_client/utils/multipart_transport.py (hunk 4 above)
Comment (Woody1193) [reply src/mms_client/utils/multipart_transport.py]: Deleted

**Review by Liam-Woods — APPROVED** [review]

**Pull request merged**